includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,481 @@
+	}
+
+	/**
+	 * WooCommerce Init
+	 */
+	public function woocommerce_init() {
+		include_once( dirname( __FILE__ ) . '/class-wc-background-swedbank-pay-queue.php' );
+		self::$background_process = new WC_Background_Swedbank_Pay_Queue();
+	}
+
+	/**
+	 * WooCommerce Loaded: load classes
+	 */
+	public function woocommerce_hook_loaded() {
+		// Includes
+		include_once( dirname( __FILE__ ) . '/class-wc-payment-token-swedbank-pay.php' );
+		include_once( dirname( __FILE__ ) . '/interfaces/class-wc-payment-gateway-swedbank-pay-interface.php' );
+		include_once( dirname( __FILE__ ) . '/abstracts/abstract-wc-payment-gateway-swedbank-pay.php' );
+	}
+
+	/**
+	 * Register payment gateway
+	 *
+	 * @param string $class_name
+	 */
+	public static function register_gateway( $class_name ) {
+		global $px_gateways;
+
+		if ( ! $px_gateways ) {
+			$px_gateways = [];
+		}
+
+		if ( ! isset( $px_gateways[ $class_name ] ) ) {
+			// Initialize instance
+			if ( $gateway = new $class_name ) {
+				$px_gateways[] = $class_name;
+
+				// Register gateway instance
+				add_filter( 'woocommerce_payment_gateways', function ( $methods ) use ( $gateway ) {
+					$methods[] = $gateway;
+
+					return $methods;
+				} );
+			}
+		}
+	}
+
+	/**
+	 * Allow processing/completed statuses for capture
+	 *
+	 * @param array $statuses
+	 * @param WC_Order $order
+	 *
+	 * @return array
+	 */
+	public function add_valid_order_statuses( $statuses, $order ) {
+		$payment_method = swedbank_pay_obj_prop( $order, 'payment_method' );
+		if ( in_array( $payment_method, self::PAYMENT_METHODS ) ) {
+			$statuses = array_merge( $statuses, [
+				'processing',
+				'completed'
+			] );
+		}
+
+		return $statuses;
+	}
+
+	/**
+	 * Order Status Change: Capture/Cancel
+	 *
+	 * @param $order_id
+	 * @param $from
+	 * @param $to
+	 * @param $order
+	 */
+	public static function order_status_changed( $order_id, $from, $to, $order ) {
+		// We are need "on-hold" only
+		if ( $from !== 'on-hold' ) {
+			return;
+		}
+
+		if ( version_compare( WC_VERSION, '3.0', '<' ) ) {
+			$order = wc_get_order( $order_id );
+		}
+
+		$payment_method = swedbank_pay_obj_prop( $order, 'payment_method' );
+		if ( ! in_array( $payment_method, self::PAYMENT_METHODS ) ) {
+			return;
+		}
+
+		/** @var WC_Payment_Gateway_Swedbank_Pay $gateway */
+		$gateway = swedbank_pay_payment_method_instance( $order );
+
+		switch ( $to ) {
+			case 'cancelled':
+				// Cancel payment
+				if ( $gateway->can_cancel( $order ) ) {
+					try {
+						swedbank_pay_cancel_payment( $order_id );
+					} catch ( Exception $e ) {
+						$message = $e->getMessage();
+						WC_Admin_Meta_Boxes::add_error( $message );
+
+						// Rollback
+						$order->update_status( $from, sprintf( __( 'Order status rollback. %s', 'swedbank-pay-woocommerce-payments' ), $message ) );
+					}
+				}
+				break;
+			case 'processing':
+			case 'completed':
+				// Capture payment
+				if ( $gateway->can_capture( $order ) ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of echo language construct is discouraged.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)